### PR TITLE
SignalR - Add more properties to the models, add usage API and complete tests

### PR DESCRIPTION
**Pull request by allxiao**
## Description

The change is based on the spec PR: https://github.com/Azure/azure-rest-api-specs/pull/3686 . We plan to release the version `0.10.0-preview` for the Ignite.

Changes:

* Add version to the `SignalRInstance` model
* Add connection strings to the `SignalRKeys` model
* Add usage operations for quota limit related query
* Add tests

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.SignalR</PackageId>
 		<Description>Provides management capabilities for Azure SignalR.</Description>
-		<Version>0.9.0-preview</Version>
+		<Version>1.0.0.0</Version>
 		<AssemblyName>Microsoft.Azure.Management.SignalR</AssemblyName>
 		<PackageTags>Microsoft Azure SignalR;SignalR;</PackageTags>
 		<PackageReleaseNotes>
```
Comment (dsgouda) [inline src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj]: Please update `PackageReleaseNotes` here

File: src/SDKs/_metadata/signalr_resource-manager.txt [hunk 2]
```diff
@@ -1,14 +1,14 @@
 ﻿Executing AutoRest command
-cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/signalr/resource-manager/readme.md --csharp --csharp-sdks-folder=C:\azure-sdk-for-net\tools\..\src\SDKs\ --version=latest --reflect-api-versions
-2018-05-03 17:51:57 UTC
+cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/signalr/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=D:\Code\azure-sdk-for-net\tools\..\src\SDKs\
+2018-08-31 09:44:19 UTC
 1) azure-rest-api-specs repository information
 GitHub fork: Azure
 Branch:      master
-Commit:      0a68675c56fceef8b5069a3026286539356feadb
```
Comment (dsgouda) [inline src/SDKs/_metadata/signalr_resource-manager.txt]: Please run msbuild build.proj and regenerate the code

File: src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>SignalR.Tests</PackageId>
+    <Description>SignalR.Tests Class Library</Description>
+    <AssemblyName>SignalR.Tests</AssemblyName>
+    <VersionPrefix>1.0.0.0</VersionPrefix>
+  </PropertyGroup>
+
+  <!--<PropertyGroup>
```
Comment (dsgouda) [inline src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj]: Remove this line

File: src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>SignalR.Tests</PackageId>
+    <Description>SignalR.Tests Class Library</Description>
+    <AssemblyName>SignalR.Tests</AssemblyName>
+    <VersionPrefix>1.0.0.0</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj]: Replace this with `<Version>1.0.0</Version>`

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj [hunk 5]
```diff
@@ -6,7 +6,7 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.SignalR</PackageId>
 		<Description>Provides management capabilities for Azure SignalR.</Description>
-		<Version>0.9.0-preview</Version>
+		<Version>1.0.0.0</Version>
```
[the remark this answers is not in the document]
Comment (ArieShout) [reply src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj]: We plan to make a GA release in September. I will update the RP to mark REST API version as stable, update the swager definition and update the code here.

[the remark this answers is not in the document]
Comment (ArieShout) [reply src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj]: Hi @dsgouda , we want to publish a stable version when the service goes to GA this September.
To confirm, in order to do that, we need:
1. update RP to accept a new stable `api-version`
2. update swager definition
3. regenerate the code in this PR and update versions if necessary.
Are these steps correct?

And if in future, we have some new REST API on boarded as preview, we need to ensure the version defined here contains a `-preview` suffix, right?

Thanks,

[the remark this answers is not in the document]
Comment (dsgouda) [reply src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj]: You got it perfectly right @ArieShout 
I suggest we close this PR and open a new one once the stable version of REST API swagger gets checked in

**Comment by ArieShout** [comment]
Hi @dsgouda , we plan to stay in preview before Ignite and work on stable api-version and SDK release after that.
Could you please help to reopen the PR? I have prepared the changes according to your review comments locally and will update after reopening.
Thanks,

File: src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj (hunk 5 above)
[the remark this answers is not in the document]
Comment (ArieShout) [reply src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj]: Thanks for confirming this. We plan to follow the steps after this preview release for Ignite.

**Comment by dsgouda** [comment]
Thanks for confirming. Reopening PR. Will review once changes are pushed

File: src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj (hunk 5 above)
[the remark this answers is not in the document]
Comment (ArieShout) [reply src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj]: Updated to `0.10.0-preview`. Versions in `AssemblyInfo.cs` were updated to `0.10.0.0`

File: src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj (hunk 1 above)
Comment (ArieShout) [reply src/SDKs/SignalR/Management.SignalR/Microsoft.Azure.Management.SignalR.csproj]: Updated.

File: src/SDKs/_metadata/signalr_resource-manager.txt (hunk 2 above)
Comment (ArieShout) [reply src/SDKs/_metadata/signalr_resource-manager.txt]: Regenerated with

```
cd tools
.\generateTool.ps1 -ResourceProvider signalr/resource-manager -PowershellInvoker
```

It seems the `generateTool.ps1` and `Start-CodeGeneration` in `tools\SdkBuildTools\psModules\CodeGenerationModules\generateDotNetSdkCode.psm1` are not passing the `ResourceProvider` parameter correctly. I need to manually update the PowerShell code to make it work.

File: src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj (hunk 3 above)
Comment (ArieShout) [reply src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj]: Removed.

File: src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj (hunk 4 above)
Comment (ArieShout) [reply src/SDKs/SignalR/SignalR.Tests/SignalR.Tests.csproj]: Updated the version.

**Comment by ArieShout** [comment]
Updated the code according to the review comments and rebased to the latest `psSdkJson6` branch.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**